groovy/groovy.editor/src/org/netbeans/modules/groovy/editor/completion/KeywordCompletion.java

```diff
@@ -98,6 +110,27 @@ public boolean complete(Map<Object, CompletionProposal> proposals, CompletionCon
         return true;
     }
 
+    boolean isFirstStatement(final CompletionContext request) {
+        TokenSequence<GroovyTokenId> ts = LexUtilities.getGroovyTokenSequence(request.doc, 1);
+
+        if (ts != null) {
+            ts.move(request.lexOffset);
+            if (ts.movePrevious()) {
+                while (ts.isValid() && ts.movePrevious() && ts.offset() >= 0) {
+                    Token<GroovyTokenId> t = ts.token();
+                    if (!(t.id() == GroovyTokenId.NLS || t.id() == GroovyTokenId.WHITESPACE 
```

Comment:
This could be moved off to some utilities a a `Predicate<Token<GroovyTokenId>>` as I fear that whitespace-traversing code is on many places and does not take all the WS/comment tokens into account (see for example the package name back traversal code in this PR).

Reply:
This is good idea. Do you want to do it in this PR?

Reply:
perhaps yes, but a possible retrofit of other code (that should use the helper instead of custom WS token checks) should be perhaps done separately. Your call.